### PR TITLE
Tsify clang-tidy-tool

### PR DESCRIPTION
**Pull request by scotran**
Tsify `clang-tidy-tool` and added optional fields to `ToolResult`.

### CODE REVIEW

**Comment by scotran** [comment]
`stdin` isn't used in the body of `runTool()`, but I kept it because it was there previously (just a note in case we want to remove now)

File: lib/tooling/clang-tidy-tool.ts [hunk 1]
```diff
@@ -75,14 +87,14 @@ export class ClangTidyTool extends BaseTool {
         // TODO: do we want compile_flags.txt rather than prefixing everything with -extra-arg=
         await fs.writeFile(path.join(dir, 'compile_flags.txt'), compileFlags.join('\n'));
         const result = await super.runTool(compilationInfo, sourcefile, args);
-        result.sourcechanged = false;
+        result.sourceChanged = false;
```
Comment (jeremy-rifkin) [inline lib/tooling/clang-tidy-tool.ts:90]: We can't make this change here without also updating code in the front-end that reads `sourcechanged`

File: lib/tooling/clang-tidy-tool.ts [hunk 2]
```diff
@@ -41,14 +42,25 @@ export class ClangTidyTool extends BaseTool {
         this.addOptionsToToolArgs = false;
     }
 
-    async runTool(compilationInfo, inputFilepath, args, stdin, supportedLibraries) {
+    override async runTool(
+        compilationInfo: Record<any, any>,
```
Comment (jeremy-rifkin) [inline lib/tooling/clang-tidy-tool.ts:46]: I think this is a `CompilationInfo`. We can defer updating this to another PR if needed (I see `Record<any, any>` is used throughout the lib/tooling code)

File: lib/tooling/clang-tidy-tool.ts [hunk 3]
```diff
@@ -41,14 +42,25 @@ export class ClangTidyTool extends BaseTool {
...
+        args?: string[],
+        stdin?: string,
+        supportedLibraries?: Record<string, Library>,
+    ) {
         const sourcefile = inputFilepath;
         const options = compilationInfo.options;
         const dir = path.dirname(sourcefile);
-        const includeflags = super.getIncludeArguments(compilationInfo.libraries, supportedLibraries);
-        const libOptions = super.getLibraryOptions(compilationInfo.libraries, supportedLibraries);
+        const includeflags = supportedLibraries
```
Comment (jeremy-rifkin) [inline lib/tooling/clang-tidy-tool.ts:55]: `compiler-dropin-tools.ts` does this as `const includeflags = this.getIncludeArguments(compilationInfo.libraries, supportedLibraries || {});` (and same with the getLibraryOptions call), they should be equivalent but we may as well try to be consistent here

**Review by jeremy-rifkin — COMMENTED** [review]
Thanks for the conversion!

File: lib/tooling/clang-tidy-tool.ts (hunk 1 above)
Comment (scotran) [reply lib/tooling/clang-tidy-tool.ts:90]: gotcha

File: lib/tooling/clang-tidy-tool.ts (hunk 2 above)
Comment (scotran) [reply lib/tooling/clang-tidy-tool.ts:46]: Yea, I think I checked out CompilationInfo but it had extra fields that wasn't used in the tooling code so chose to stick with the current `Record<any, any>`

File: lib/tooling/clang-tidy-tool.ts (hunk 3 above)
Comment (scotran) [reply lib/tooling/clang-tidy-tool.ts:55]: 👍

File: lib/tooling/clang-tidy-tool.ts (hunk 1 above)
Comment (scotran) [reply lib/tooling/clang-tidy-tool.ts:90]: for simplicity, we can just leave it sourcechanged for now? unless you think it'd be better to change

Comment (jeremy-rifkin) [reply lib/tooling/clang-tidy-tool.ts:90]: Yes, let's leave it as sourcechanged/newsource for now

**Review by jeremy-rifkin — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**